build-scripts/build_tests.py

```diff
@@ -0,0 +1,237 @@
+                logger.debug("Template %s doesn't have tests. Skipping for rule %s.",
+                             template_name, rule_id)
+                continue
+            test_config_path = rule_tests_root / "test_config.yml"
+            deny_templated_scenarios = _get_deny_templated_scenarios(test_config_path)
+            for test in template_tests_root.iterdir():  # type: pathlib.Path
+                if not test.name.endswith(".sh") or test.name in deny_templated_scenarios:
+                    logging.warning("Skipping % for %s", test.name, rule_id)
+                    continue
+                template = ssg.templates.Template.load_template(str(templates_root.absolute()),
+                                                                template_name)
+                rendered_rule_obj["template"]["vars"]["_rule_id"] = rule_id
+                template_parameters = template.preprocess(
+                    rendered_rule_obj["template"]["vars"], "test")
+                env_yaml = env_yaml.copy()
+                jinja_dict = ssg.utils.merge_dicts(env_yaml, template_parameters)
+                file_contents = ssg.jinja.process_file_with_macros(str(test.absolute()),
+                                                                   jinja_dict)
+                scenario = tests.ssg_test_suite.rule.Scenario(test.name, file_contents)
+                if scenario.matches_platform(benchmark_cpes):
+                    rule_output_path.mkdir(parents=True, exist_ok=True)
+                    test_output_path = rule_output_path / test.name
+                    _write_path(file_contents, test_output_path)
+                    logger.debug('Wrote scenario %s for rule %s', test.name, rule_id)
+                else:
+                    logger.debug('Skipping scenario %s for rule %s has it not applicable',
+                                 test.name, rule_id)
+
+
+def _get_benchmark_cpes(env_yaml):
+    benchmark_cpes = set()
+    for cpe in env_yaml["cpes"]:
+        for _, data in cpe.items():
+            benchmark_cpes.add(data["name"])
+    return benchmark_cpes
+
+
+def _get_rules_in_profile(built_profiles_root):
+    rules_in_profile = list()
+    for profile_file in built_profiles_root.iterdir():  # type: pathlib.Path
+        if not profile_file.name.endswith('.profile'):
+            continue
+        profile_data = ssg.yaml.open_raw(str(profile_file.absolute()))
+        for selection in profile_data["selections"]:
+            if '=' not in selection:
+                rules_in_profile.append(selection)
+    return rules_in_profile
+
+
+def main() -> int:
+    args = _create_arg_parser().parse_args()
+    logging.basicConfig(level=logging.getLevelName(args.log_level))
+    env_yaml = ssg.environment.open_environment(
+        args.build_config_yaml, args.product_yaml)
+
+    root_path = pathlib.Path(args.root).resolve().absolute()
+    output_path = pathlib.Path(args.output).resolve().absolute()
```

Comment:
`.resolve()` already makes the path absolute:
```
$ pydoc3 pathlib.Path.resolve | cat
Help on function resolve in pathlib.Path:

pathlib.Path.resolve = resolve(self, strict=False)
    Make the path absolute, resolving all symlinks on the way and also
    normalizing it.
```

Reply:
I can clean that up.